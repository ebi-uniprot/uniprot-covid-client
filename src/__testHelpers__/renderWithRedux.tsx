/* eslint-disable */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import rootReducer from '../state/rootReducer';
import { Router, Route } from 'react-router-dom';

type RenderOptions = {
  route?: string;
  history?: MemoryHistory<any>;
  path?: string;
  initialState?: any;
  store?: any;
};

const renderWithRedux = (
  ui: React.Component,
  {
    route = '',
    history = createMemoryHistory({ initialEntries: [route] }),
    path,
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  }: RenderOptions = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          {path ? <Route path={path} render={() => ui} /> : ui}
        </Router>
      </Provider>
    ),
    store,
    history,
  };
};

export default renderWithRedux;
