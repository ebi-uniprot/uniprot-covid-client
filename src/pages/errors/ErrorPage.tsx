import React from 'react';

import '../../styles/ErrorPages.scss';

const ErrorPage: React.FC<{
  artwork: JSX.Element;
  message: JSX.Element;
  testid?: string;
}> = ({ artwork, message, testid }) => (
  <div className="error-page-container" data-testid={testid || ''}>
    <artwork.type
      {...artwork.props}
      className="error-page-container__art-work"
    />
    {message}
  </div>
);

export default ErrorPage;
