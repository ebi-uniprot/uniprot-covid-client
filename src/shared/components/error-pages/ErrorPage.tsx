import React, { ReactNode, ReactElement } from 'react';

import './styles/error-pages.scss';

const ErrorPage: React.FC<{
  artwork: ReactElement;
  message: ReactNode;
  testid?: string;
}> = ({ artwork, message, testid }) => (
  <div className="error-page-container" data-testid={testid || ''}>
    <artwork.type
      {...artwork.props}
      className="error-page-container__art-work"
    />
    {message}
  </div>
);

export default ErrorPage;
