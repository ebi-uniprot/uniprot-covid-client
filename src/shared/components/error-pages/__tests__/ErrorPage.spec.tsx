import React from 'react';
import ErrorPage from '../ErrorPage';
import renderWithRedux from '../../../__test-helpers__/RenderWithRedux';

import ArtWork from '../svgs/error.svg';

describe('ErrorPage component', () => {
  test('should render', async () => {
    const { asFragment, queryByText } = renderWithRedux(
      <ErrorPage artwork={<ArtWork />} message="test message" testid="testid" />
    );
    expect(await queryByText('test message')).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });
});
