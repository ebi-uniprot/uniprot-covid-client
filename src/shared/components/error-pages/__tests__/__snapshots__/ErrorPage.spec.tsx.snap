// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage component should render 1`] = `
<DocumentFragment>
  <div
    class="error-page-container"
    data-testid="testid"
  >
    <test-file-stub
      classname="error-page-container__art-work"
    />
    test message
  </div>
</DocumentFragment>
`;
